### ISSUE TITLE
Can not get esbuild configuration working, after migrating from `TheBrainFamily/cypress-cucumber-preprocessor`

### ISSUE THREAD

**Issue by Bertg** [badeball/cypress-cucumber-preprocessor#777]
### Current behavior

I'm trying to migrate to this version of `cypress-cucumber-preprocessor` using the recommended esbuild method. However, I can't seem to get it working at all. 

What is special about our setup is that our cypress setup is independent from the application it is testing. This should not be a problem as we only use it for e2e testing. This however does mean, that a lot of assumed configurations or packages are not presenting our setup.

```
# package.json
{
  "name": "integrations",
  "license": "UNLICENSED",
  "private": true,
  "description": "Integration testing using Cypress",
  "main": "index.js",
  "scripts": {
    "open": "cypress open",
    "test": "cypress run"
  },
  "dependencies": {
    "@badeball/cypress-cucumber-preprocessor": "latest",
    "@bahmutov/cypress-esbuild-preprocessor": "latest",
    "@esbuild-plugins/node-modules-polyfill": "latest",
    "cypress": "latest",
    "cypress-file-upload": "latest",
    "cypress-iframe": "latest",
    "cypress-otp": "latest",
    "esbuild": "latest",
    "uuid": "latest"
  },
  "cypress-cucumber-preprocessor": {
    "nonGlobalStepDefinitions": true,
    "stepDefinitions": [
      "[filepath]/steps.{js,ts}"
    ]
  }
}

```

```
# cypress config

const { defineConfig } = require("cypress");

module.exports = defineConfig({
  video: false,
  // chromeWebSecurity: false,
  defaultCommandTimeout: 10000,
  env: {
    TAGS: 'not (@ci-broken or @unstable)',
  },
  retries: {
    runMode: 2,
    openMode: 0,
  },
  e2e: {
    setupNodeEvents: require('./cypress/plugins/index.js'),
    specPattern: 'cypress/e2e/**/*.{feature,features,spec.js}',
    baseUrl: 'https://app.localtest.me',
  },
})
```

```
# cypress/plugins/index.js
/// <reference types="cypress" />
// ***********************************************************
// This example plugins/index.js can be used to load plugins
//
// You can change the location of this file or turn off loading
// the plugins file with the 'pluginsFile' configuration option.
//
// You can read more here:
// https://on.cypress.io/plugins-guide
// ***********************************************************

const createBundler = require("@bahmutov/cypress-esbuild-preprocessor");
const preprocessor = require("@badeball/cypress-cucumber-preprocessor");
const createEsbuildPlugin = require("@badeball/cypress-cucumber-preprocessor/esbuild");
// const nodePolyfills = require('@esbuild-plugins/node-modules-polyfill').NodeModulesPolyfillPlugin;
const generateOTP = require('cypress-otp');

// This function is called when a project is opened or re-opened (e.g. due to
// the project's config changing)

/**
 * @type {Cypress.PluginConfig}
 */
module.exports = async function setupNodeEvents(on, config) {
  await preprocessor.addCucumberPreprocessorPlugin(on, config);

  on(
    "file:preprocessor",
    createBundler({
      plugins: [
        // nodePolyfills(),
        createEsbuildPlugin.default(config),
      ],
    })
  );

  on('task', { generateOTP });

  // Make sure to return the config object as it might have been modified by the plugin.
  return config;
}

```

There are no other config files.

When running this, cypress keeps crashing with this error:

```
Build failed with 6 errors:
node_modules/esbuild/lib/main.js:1709:17: ERROR: Could not resolve "fs"
node_modules/esbuild/lib/main.js:1710:17: ERROR: Could not resolve "os"
node_modules/esbuild/lib/main.js:1711:19: ERROR: Could not resolve "path"
node_modules/esbuild/lib/main.js:1858:28: ERROR: Could not resolve "child_process"
node_modules/esbuild/lib/main.js:1859:21: ERROR: Could not resolve "crypto"
```

When I try to run it with `nodePolyfills` [as described here](https://stackoverflow.com/a/72896285), we get an other error:

```
process is not defined
```

Which leads me further down a rabbit hole, that I think is the wrong rabbit hole.

### Desired behavior

Get cypress running with `@badeball/cypress-cucumber-preprocessor` using the suggested `esbuild-cjs` approach.

### Test code to reproduce

See code above.

### Versions

* **Cypress version**: 10.3.1
* **Preprocessor version**: 11.5.1
* **Node version**: 16.14.2

### Checklist

- [x] I've read the [FAQ](https://github.com/badeball/cypress-cucumber-preprocessor/blob/master/docs/faq.md).
- [x] I've read [Instructions for logging issues](https://github.com/badeball/cypress-cucumber-preprocessor/blob/master/CONTRIBUTING.md#instructions-for-logging-issues).
- [x] I'm not using `cypress-cucumber-preprocessor@4.3.1` (package name has changed and it is no longer the most recent version, see [#689](https://github.com/badeball/cypress-cucumber-preprocessor/issues/689)).

**Comment by badeball** [comment]
I can't reproduce the issue with the provided information alone. I suspect that there's something going on with your actual step definitions. In any case, you will have to provide me a cloneable repository that allows me to observe the error.

**Comment by Bertg** [comment]
I'm so very sorry, I'm hitting myself fin the head right now.

As you suggested, I created a example repo, and in copying over the essential files (one of which I didn't mention in the original ticket) I noticed the following issue in a common step file.

```js
import { Given, Then } from "@bahmutov/cypress-esbuild-preprocessor";
```

Which obviously should be...

```js
import { Given, Then } from "@badeball/cypress-cucumber-preprocessor";
```

Complete oversight from me, I should have payed closer attention. This issue existed a few times in the code, and I guess I didn't notice due to the similarities of the package names.

Terribly sorry for wasting your time, but thanks for responding promptly.
